pandas/core/groupby/generic.py

```diff
@@ -526,7 +526,7 @@ def _transform_fast(self, result, func_nm: str) -> Series:
         cast = self._transform_should_cast(func_nm)
         out = algorithms.take_1d(result._values, ids)
         if cast:
-            out = self._try_cast(out, self.obj)
+            out = self._try_cast(out, self.obj, how=func_nm)
```

Comment:
you can simply put what you have not in group_result_dtype into _try_cast, IOW if its a boolean, then you can just change the dtype that's passed to maybe_downcast_numeric